Hestia/app/src/main/java/hestia/backend/Cache.java

```diff
@@ -0,0 +1,95 @@
+package hestia.backend;
+
+import java.util.ArrayList;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+public class Cache {
+    /**
+     * We use a CopyOnWriteArrayList to avoid ConcurrentModificationExceptions if
+     * a listener attempts to remove itself during event notification.
+     */
+    private final CopyOnWriteArrayList<DevicesChangeListener> listeners =
+            new CopyOnWriteArrayList<>();
+    private static Cache instance;
+    private ArrayList<Device> devices = new ArrayList<>();
+    private final static String TAG = "Cache";
+
+    private String ip = "82.73.173.179";
+    private int port = 8000;
+
+    private Cache() {}
```

Comment:
Maybe add a line here as well to explain why we need a private default constructor